### ISSUE TITLE
“Share” button with an insufficient accessible name [iOS]

### ISSUE THREAD

**Issue by willslone** [covid-alert-ny/covid-green-app#190]
There is a graphical button on numerous screens that opens the typical “share” activity view commonly seen in iOS applications. The button has an accessible name of “View or amend application settings”, which suggests that the button takes you to a view where specific application settings can be viewed/changed.

![Graphical button focused with VoiceOver and announced as “View or amend application settings”](https://user-images.githubusercontent.com/4437084/91862014-721dc280-ec3b-11ea-80d4-b866a68fb918.png)

There is a button with the text label of “Share” on the Onboarding Confirm success screen that opens the same view. Generally, controls that trigger the same action should be identified consistently across the application.

Use the accessibilityLabel attribute on the image to give it a proper accessible name.

```javascript
<Image source={…} accessibilityLabel='Share' />
```

accessibilityLabel can also be used on the container element of the image (e.g., TouchableOpacity, Pressable, etc.). If accessibilityLabel is used both on the image and the container element, the accessibilityLabel of the container element will be announced to VoiceOver users.

```javascript
<TouchableOpacity onPress={…} accessibilityLabel='Share' accessibilityRole='button'>
  <Image source={…} />
</TouchableOpacity>
```

**Comment by willslone** [comment]
I retested and can confirm the issue has been resolved.
